onnx/backend/test.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import unittest
+
+import numpy as np
+
+import onnx
+from onnx import helper
+from .test_util import create_input, N
+
+L = 20
+M = 10
+S = 5
+const2_np = np.random.randn(S, S)
+const2_onnx = onnx.helper.make_tensor("const2",
+                                      onnx.TensorProto.FLOAT,
+                                      (S, S),
+                                      const2_np.flatten().astype(float))
+
+# TODO: These Numpy specs will be generally useful to backend implementations,
+# so they should get moved out of here at some point
+node_tests = [
+  ("test_abs", N("Abs"), np.abs, [(S, S, S)]),
+  ("test_add", N("Add"), np.add, [(S, S, S), (S, S, S)]),
+  ("test_add_bcast", N("Add", broadcast=1), np.add, [(S, M), (M,)]),
+  ("test_constant", N("Constant", value=const2_onnx), lambda: const2_np, []),
+  # TODO: Are we actually supporting other dot modes?  In that case, some fancy
+  # footwork is necessary...
+  ("test_dot", N("Dot"), np.dot, [(S, M), (M, L)]),
+  ("test_relu", N("Relu"), lambda x: np.clip(x, 0, np.inf), [(S, S, S)]),
+  # TODO: Add all the other operators
+]
```

Comment:
Note: It's going to be a bit difficult to write tests for things like convolution using bare numpy.  One good approach is to use tensor comprehensions but we still need a reference implementation of TCs we can roll into ONNX.